src/yarpWholeBodyStates.cpp

```diff
@@ -760,7 +768,37 @@ bool yarpWholeBodyEstimator::threadInit()
     left_sole_local_id = wbi::ID(LEFT_LEG,8);
     right_sole_local_id = wbi::ID(RIGHT_LEG,8);
     */
-
+    
+    /*    Eigen::Map<Eigen::VectorXd> dqjVect(dqj,dof);
+    Eigen::Matrix<double,6,Eigen::Dynamic,Eigen::RowMajor> complete_jacobian(6,dof+6), joint_jacobian(6,dof),floatingBase_jacobian(6,6);
+    Eigen::Matrix<double,6,Eigen::Dynamic,Eigen::RowMajor> tempMatForComputation(6,dof);//, minusTemp(6,dof);
+    Eigen::VectorXd dvbVect(6);
+    Eigen::Map<Eigen::VectorXd> yarpWrapper(estimates.lastBaseVel.data(), estimates.lastBaseVel.size());
+  */  
+  
+     int dof = estimates.lastQ.length();
+//     dqjVect_.resize(6,dof+6);
+//     std::cout<<" dqjVect : "<<dqjVect_.rows()<<" , "<<dqjVect_.cols()<<"\n"; 
+
+
+    
+    complete_jacobian.resize(6,dof+6);
+    joint_jacobian.resize(6,dof);
+//     floatingBase_jacobian(6,6);
+    tempMatForComputation.resize(6,dof);
+    new (&rotationalVelocityWrapper) Eigen::Map<Eigen::VectorXd>(estimates.lastBaseVel.data(), estimates.lastBaseVel.size());
+    new (&dqjVect) Eigen::Map<Eigen::VectorXd>(estimates.lastDq.data(),dof);
```

Comment:
Yes I was following this :
http://eigen.tuxfamily.org/dox/group__TutorialMapClass.html
The last example has it in that form

---
Naveen Kuppuswamy, PhD
Post-doctoral Fellow,
Cognitive Humanoids Lab,
Department of Robotics, Brain and Cognitive Sciences (RBCS),
Istituto Italiano di Tecnologia, Genova, Italy

On Tue, Mar 24, 2015 at 4:40 PM, Silvio Traversaro <notifications@github.com
> wrote:

> In src/yarpWholeBodyStates.cpp
> <https://github.com/robotology-playground/yarp-wholebodyinterface/pull/23#discussion_r27040006>
> :
>
> > +    Eigen::VectorXd dvbVect(6);
> > +    Eigen::Map<Eigen::VectorXd> yarpWrapper(estimates.lastBaseVel.data(), estimates.lastBaseVel.size());
> > +  */
> > +
> > +     int dof = estimates.lastQ.length();
> > +//     dqjVect_.resize(6,dof+6);
> > +//     std::cout<<" dqjVect : "<<dqjVect_.rows()<<" , "<<dqjVect_.cols()<<"\n";
> > +
> > +
> > +
> > +    complete_jacobian.resize(6,dof+6);
> > +    joint_jacobian.resize(6,dof);
> > +//     floatingBase_jacobian(6,6);
> > +    tempMatForComputation.resize(6,dof);
> > +    new (&rotationalVelocityWrapper) Eigen::Map<Eigen::VectorXd>(estimates.lastBaseVel.data(), estimates.lastBaseVel.size());
> > +    new (&dqjVect) Eigen::Map<Eigen::VectorXd>(estimates.lastDq.data(),dof);
>
> Ah... ok! I still guess that a local Eigen::Map is more readable, but
> apparently this is endorsed on Eigen docs..
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/robotology-playground/yarp-wholebodyinterface/pull/23/files#r27040006>
> .
>